### DISCUSSION TITLE
Override and Accessibility

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7258] [Q&A] [answered]
Currently, when you override a member, you must restate the accessibility of the member (public, protected, etc).

Is this necessarily required? Could the language be enhanced to make this optional?

**Comment by HaloFour** [comment, marked as answer]
I prefer the clarity.  It's something the IDE will insert for you anyway.  Also, the runtime doesn't require that the accessibility of overrides match the overridden member, so making it implicit could impact future language design to make overrides more flexible.

**Comment by TahirAhmadov** [comment]
I kind of like C++ style accessibility regions, actually. I tend to group members by accessibility and category anyway, like public methods, public properties, etc. However, the C++ approach can be a little difficult to read in larger classes - if can be difficult to easily tell which accessibility region a given members happens to be in. Maybe it would be better with an explicit curly-brace-delimited region? That would create another indentation level, but would save a lot of the keyword clutter. Anyway, I doubt this would ever happen in C#.
```cs
class Foo
{
  public
  {
    void Bar() { ... }
  }
}
```

**Reply by theunrepentantgeek** [reply]
Delphi used visibility regions as you describe, conventionally listed in the order **published**, **public**, **protected** and **private**. They had a side effect of often enforcing the separation of related members with different visibilities. I missed them briefly when I started working with C#, but have never lamented their lack.

(My experience was with Delphi versions 1-7; I don't know if later versions relaxed the rules/conventions.)

**Reply by TahirAhmadov** [reply]
Yeah, I'm not annoyed by C# accessibility keyword rules myself, very close to the bottom of my priority list of things to change. I actually never have members with different visibilities next to each other, but I guess that's just my preferred coding style.

**Reply by DavidArno** [reply]
The only one that I'd change if I could is the need to repeat `static` for members in a static class. Certainly doesn't annoy me though. And arguably isn't an  accessibility keyword anyway 😆

**Reply by TahirAhmadov** [reply]
@DavidArno something like:
```cs
class Foo
{
  public static // etc.
  {
    void Bar() { ... }
  }
}
```
